data_engineering.qmd

```diff
@@ -1,121 +1,526 @@
+
+**Risk Management:** Data version control allows transparency and accountability by tracking versions of the dataset.
+
+**Collaboration and Efficiency:** Easy access to different versions of the dataset in one place can improve data sharing of specific checkpoints, and enable efficient collaboration.
+
+**Reproducibility:** Data version control allows for tracking the performance of models with respect to different versions of the data,
+and therefore enabling reproducibility.
+
+**Key Concepts**
+
+-   *Commits:* It is an immutable snapshot of the data at a specific point in time, representing a unique version. Every commit is associated with a unique identifier to allow
+
+-   *Branches*: Branching allows developers and data scientists to diverge from the main line of development and continue to work independently without affecting other branches. This is especially useful when experimenting with new features or models, enabling parallel development and experimentation without the risk of corrupting the stable, main branch.
+
+-   *Merges:* Merges help to integrate changes from different branches while maintaining the integrity of the data.
+
+**Popular Data Version Control Systems**
+
+[**[DVC]{.underline}**](https://dvc.org/doc): It stands for Data Version Control in short, and is an open-source, lightweight tool that works on top of github and supports all kinds of data format. It can seamlessly integrate into the Git workflow, if Git is being used for managing code. It captures the versions of data and models in the Git commits, while storing them on premises or on cloud (e.g. AWS, Google Cloud, Azure). These data and models (e.g. ML artifacts) are defined in the metadata files, which get updated in every commit. It can allow metrics tracking of models on different versions of the data.
+
+**[[lakeFS]{.underline}](https://docs.lakefs.io/):** It is an open-source tool that supports the data version control on data lakes. It supports many git-like operations such as branching and merging of data, as well as reverting to previous versions of the data. It also has a unique UI feature which allows exploration and management of data much easier.
+
+**[[Git LFS]{.underline}](https://git-lfs.com/):** It is useful for data version control on smaller sized datasets. It uses Git's inbuilt branching and merging features, but is limited in terms of tracking metrics, reverting to previous versions or integration with data lakes.
 
-- Version Control Systems 
-- Metadata 
 
 ## Optimizing Data for Embedded AI
```

Comment:
Honestly, I'm not sure how much this section adds as a standalone section vs as part of the prior section?